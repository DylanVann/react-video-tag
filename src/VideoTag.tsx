import React, { SFC } from 'react'
import { videoTagString, VideoTagProps } from './videoTagString'

export const VideoTag: SFC<VideoTagProps> = props => (
    <span
        dangerouslySetInnerHTML={{
            __html: videoTagString(props),
        }}
    />
)

VideoTag.defaultProps = {
    muted: true,
    autoPlay: true,
    playsInline: true,
    loop: true,
}
