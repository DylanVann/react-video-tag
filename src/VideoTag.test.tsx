import React from 'react'
import renderer from 'react-test-renderer'
import { VideoTag } from './VideoTag'

it('renders', () => {
    const tree = renderer.create(<VideoTag />).toJSON()
    expect(tree).toMatchSnapshot()
})

it('renders with src', () => {
    const tree = renderer.create(<VideoTag src="test.png" />).toJSON()
    expect(tree).toMatchSnapshot()
})
