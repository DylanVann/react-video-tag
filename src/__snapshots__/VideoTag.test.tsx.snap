// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<video muted autoplay playsinline loop><source src=\\"\\"></video>",
    }
  }
/>
`;

exports[`renders with src 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<video muted autoplay playsinline loop><source src=\\"test.png\\"></video>",
    }
  }
/>
`;
